core/tests/protractor_utils/ExplorationEditorTranslationTab.js

```diff
@@ -388,130 +364,137 @@ var ExplorationEditorTranslationTab = function() {
     }
   };
 
   this.expectSolutionTabContentToMatch = async function(content) {
-    await waitFor.elementToBeClickable(
-      solutionTabButton, 'Solution Tab button is not clickable');
-    await solutionTabButton.click();
-    expect(await solutionTabText.getText()).toMatch(content);
+    await action.click('Solution tab button', solutionTabButton);
+    expect(await action.getText(
+      'Solution tab button',
+      solutionTabText)).toMatch(content);
   };
 
   this.expectNumericalStatusToMatch = async function(content) {
-    expect(await numericalStatus.getText()).toMatch(content);
+    expect(await action.getText(
+      'Numerical status',
+      numericalStatus)).toMatch(content);
   };
 
   this.expectNumericalStatusAccessibilityToMatch = async function(content) {
-    await waitFor.visibilityOf(
-      numericalStatus,
-      'Numerical status element taking too long to appear');
-    expect(await numericalStatus.getAttribute('aria-label')).toMatch(content);
+    expect(await action.getAttribute(
+      'Numerical status element',
+      numericalStatus, 'aria-label')).toMatch(content);
   };
 
   this.expectContentAccessibilityToMatch = async function(content) {
-    expect(await translationTabContentAccessibility.getAttribute(
+    expect(await action.getAttribute(
+      'Translation tab content',
+      translationTabContentAccessibility,
       'aria-label')).toMatch(content);
   };
 
   this.expectFeedbackAccessibilityToMatch = async function(content) {
-    expect(await translationTabFeedbackAccessibility.getAttribute(
+    expect(await action.getAttribute(
+      'Translation tab feedback',
+      translationTabFeedbackAccessibility,
       'aria-label')).toMatch(content);
   };
 
   this.expectHintAccessibilityToMatch = async function(content) {
-    expect(await translationTabHintAccessibility.getAttribute(
+    expect(await action.getAttribute(
+      'Translation tab hint',
+      translationTabHintAccessibility,
       'aria-label')).toMatch(content);
   };
 
   this.expectSolutionAccessibilityToMatch = async function(content) {
-    expect(await translationTabSolutionAccessibility.getAttribute(
+    expect(await action.getAttribute(
+      'Translation tab solution',
+      translationTabSolutionAccessibility,
       'aria-label')).toMatch(content);
   };
 
   this.expectStartRecordingAccessibilityToMatch = async function(content) {
-    expect(await translationTabStartRecordingAccessibility.getAttribute(
+    expect(await action.getAttribute(
+      'Translation tab start recording',
+      translationTabStartRecordingAccessibility,
       'aria-label')).toMatch(content);
   };
 
   this.expectUploadRecordingAccessibilityToMatch = async function(content) {
-    expect(await translationTabUploadRecordingAccessibility.getAttribute(
+    expect(await action.getAttribute(
+      'Translation tab upload recording',
+      translationTabUploadRecordingAccessibility,
       'aria-label')).toMatch(content);
   };
 
```

Comment:
Can you use `action.getAttribute in `expectPlayRecordingAccessibilityToMatch()` below too?

Reply:
When I tried using it but tests fail

![Screenshot from 2021-09-14 23-58-38](https://user-images.githubusercontent.com/42772952/133313453-21181bca-fca3-4b1a-923f-a1fa5ae992fd.png)

Reply:
Interesting. Maybe the element is invisible but still has the aria attribute. In that case, using waitFor.presenceOf in `action.getAttribute` would make sense so that we correctly handle invisible elements with attributes

Reply:
Ya that's why we can't use getAttribute here

Reply:
You can change `action.getAttribute` to use `waitFor.presenceOf` instead, right?